### PR TITLE
chore: upgrade cometbft and cosmos-sdk

### PR DESCRIPTION
**Pull request by gartnera**
# Description

Upgrade cosmos-sdk and cometbft to latest minor versions.

- Close most outstanding security issues (even if they are minor/irrelevent)
- Unblock https://github.com/zeta-chain/ethermint/issues/86 because x/exp dependency was removed from cosmos/gogoproto

We have to switch to using CGO secp256k1 to bypass https://github.com/btcsuite/btcd/issues/2243 and we cannot use native btcec secp256k1 until we upgrade to cosmos-sdk v0.50.x. This doesn't really matter as our release binaries have always been built with CGO enabled.

`libsecp256k1_sdk` enables cosmos-sdk to use the same secp256k1 logic that go-ethereum uses.

This should also get us in a much better position for the cosmos-sdk v0.50.x upgrade.

We also have the option of only upgrading cosmos-sdk and ignoring the cometbft upgrade for now which should unblock https://github.com/zeta-chain/ethermint/issues/86 (but not resolve any of the cometbft security issues).

TODO:
- [x] Is a way to get the new `rpcimportable` test to work without the `libsecp256k1_sdk` tag. Update: no, there are too many dependencies on on `github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1`. Graph:
rpcimportable_secp256k1

# How Has This Been Tested?

- [ ] Tested CCTX in localnet
- [ ] Tested in development environment
- [ ] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions

## Summary by CodeRabbit

- **New Features**
	- Introduced a new CI job `ci-ok` for enhanced error handling in the testing workflow.
	- Added a new shell script `test.sh` for running tests in the `rpcimportable` module.

- **Bug Fixes**
	- Updated the sorting mechanism in the `ChainRegistry` for improved clarity and consistency.

- **Documentation**
	- Adjusted the documentation generation script for `zetacored`, removing the installation command and streamlining the process.

- **Chores**
	- Updated various dependency versions for better stability and performance.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -115,7 +115,7 @@ jobs:
         run: cat go.mod
       - name: go test
         working-directory: contrib/rpcimportable
-        run: go test ./...
+        run: ./test.sh
```
[the remark this answers is not in the document]
Comment (gartnera) [reply .github/workflows/ci.yml:118]: this is not correct

File: contrib/rpcimportable/test.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+# run rpcimportable test with the required build flags
+
+set -eo pipefail
+
+cd "$(dirname "$0")"
```
[the remark this answers is not in the document]
Comment (gartnera) [reply contrib/rpcimportable/test.sh:7]: I don't think this can ever fail

File: go.mod [hunk 3]
```diff
@@ -341,8 +340,14 @@ require (
 
 require (
 	github.com/decred/dcrd/crypto/blake256 v1.0.1 // indirect
+	github.com/hashicorp/golang-lru/v2 v2.0.7 // indirect
```
[the remark this answers is not in the document]
Comment (gartnera) [reply go.mod:343]: Importing different major versions is allowed in go

File: go.mod [hunk 4]
```diff
@@ -304,27 +303,27 @@
...
 	golang.org/x/time v0.5.0 // indirect
 	golang.org/x/tools v0.21.1-0.20240508182429-e35e4ccd0d2d // indirect
 	gonum.org/v1/gonum v0.13.0 // indirect
-	google.golang.org/api v0.152.0 // indirect
+	google.golang.org/api v0.171.0 // indirect
 	google.golang.org/appengine v1.6.8 // indirect
-	google.golang.org/genproto v0.0.0-20240102182953-50ed04b92917 // indirect
-	google.golang.org/genproto/googleapis/rpc v0.0.0-20240108191215-35c7eff3a6b1 // indirect
+	google.golang.org/genproto v0.0.0-20240213162025-012b6fc9bca9 // indirect
+	google.golang.org/genproto/googleapis/rpc v0.0.0-20240314234333-6e1732d8331c // indirect
```
[the remark this answers is not in the document]
Comment (gartnera) [reply go.mod:326]: This package does not have tagged releases

**Review by swift1337 — APPROVED** [review]

File: contrib/rpcimportable/test.sh [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+
+go test -tags libsecp256k1_sdk -ldflags=all="-extldflags=-Wl,--allow-multiple-definition" ./...
```
Comment (lumtis) [inline contrib/rpcimportable/test.sh:9]: I agree with Coderabbit. Would make sense to explain these flags in comments

**Review by lumtis — APPROVED** [review]

**Review by julianrubino — APPROVED** [review]

**Review by CryptoFewka — APPROVED** [review]
CI changes look good to me

**Pull request merged**